lib/unique_elements.rb

```diff
@@ -1,6 +1,26 @@
 # This method takes an array of elements (these might be strings, integers, floats, or a
 # combination of the above), and returns an array of the elements that appear once and only
 # once.
-def find_unique_elements(arr)
+
+def count_elements(arr)
+  h={}
+  arr.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+  end
+  return h
+end  
   
+def find_unique_elements(arr)
+    store_unique = []
+    h = count_elements(arr)
+    #next line should create an array that stores the hash keys with a value of '1' from the hash created by the 'count_elements' method
+    h.each do |k,v|
+    if h.index(1) == true
+       store_unique >> "#{k}"
+    end
```

Comment:
This code `h.index(1) == true` won't serve you well in this problem. I'll explain below, but first let's assume (for the sake of explanation) that `h` is defined as this:

```ruby
h = {
  "banana" => 4,
  "apple" => 2,
  "fig" => 1,
  "strawberry" => 49,
  "date" => 1,
  "pear" => 3
}
```

`h.index(1) == true` has nothing to do wth the current key/value pair that is being evaluated. Consider the following table as a representation of your loop through the hash:

`k` | `v` | `store_unique` | Line of Code | Notes
-------- | ------- | ------------- | --------------- | ------------------
- | - | Starts out as an empty array | - | - |
"banana" | 4 | [] | `if h.index(1) == true` | The conditional doesn't deal with either "banana" or `4`, but it's nevertheless `true` because `h.index(1)` merely checks if `h` has _any_ key in it with a value of `1`.
"banana" | 4| [] | `store_unique >> "#{k}"` | Since the conditional was true, this line will run, which adds the current key ("banana") to the `store_unique` array.
"banana" | 4 | ["banana"] | - | No more code in this block, so Ruby moves to the next key/value pair. Notice that we're already in trouble, because "banana" is not unique. But it was added because the conditional check wasn't checking the currently iterating value was `1`–it was just checking if `h` has any values of `1` at all.
"apple" | 2 | ["banana"] | `h.index(1) == true` | Same as above. This will return `true`, so the next line will run:
"apple" | 2 | ["banana"] | `store_unique >> "#{k}"` | And now "apple" gets added.

You can probably see how this will end up adding all of the hash's elements to the array. So the root problem here is the conditional `h.index(1) == true` which is simply asking the wrong question of the hash. Instead, you want to check if the **value** (`v`) for the current key/value pair is 1.